docs/one_collect-design.md

```diff
@@ -0,0 +1,651 @@
+    })
+});
+
+// 4. Start collection
+let machine = exporter.start_collection("profiling_session")?;
+
+// 5. Collection runs automatically...
+std::thread::sleep(Duration::from_secs(10));
+
+// 6. Stop and export
+exporter.stop_and_export()?;
+```
+
+##### Scenario-Based Model
+
+The Universal Export Framework moves beyond individual event specification to a scenario-based approach:
+
+```rust
+// Instead of manually specifying events:
+// session.add_event(timer_event);
+// session.add_event(context_switch_event);
+// session.add_event(process_event);
+// ... dozens more
+
+// Simply enable scenarios:
+settings.add_scenario("cpu_profiling");           // CPU sampling + stacks
+settings.add_scenario("memory_tracking");         // Heap allocations
+settings.add_scenario("io_monitoring");           // File/network I/O
+settings.add_scenario("dotnet_gc_analysis");      // .NET GC events
+
+// Each scenario automatically determines required events
+// and handles cross-platform differences
+```
+
+#### Helpers System (`helpers` module)
+
+##### Callstack Helper (`helpers::callstack`)
+Stack unwinding integration:
+- Integrates with `ruwind` library
+- Manages unwinding context and state
+- Provides callstack symbolization
+
+##### .NET Helper (`helpers::dotnet`)
+Managed code profiling support:
+- CLR event processing
+- Managed/native boundary handling
+- .NET-specific data extraction
+
+##### Scripting Integration (`helpers::scripting`)
+
+The scripting engine integrates at the universal layer, allowing runtime customization of data processing. It hooks into the `ExportMachine` after all data has been aggregated:
+
+**Code Reference**: [`one_collect/src/helpers/exporting/scripting.rs`](one_collect/src/helpers/exporting/scripting.rs)
+
+```rust
+use one_collect::helpers::exporting::*;
+
+// Scripting hooks into the universal layer
+let mut exporter = UniversalExporter::new(settings);
+
+// Add script that processes aggregated data
+exporter.add_parsed_hook(|context: &mut UniversalParsedContext| {
+    let machine = context.machine_mut();
+    
+    // Script has access to all aggregated data
+    // and can perform custom analysis/filtering
+    machine.apply_custom_filter(|record| {
+        // Custom filtering logic
+        true
+    })?;
+    
+    machine.add_custom_metric("my_metric", calculate_metric(&machine))?;
+    Ok(())
+});
+
+// Result is an ExportMachine with all data aggregated
+// and custom processing applied
+let final_machine = exporter.process_with_scripts()?;
+```
+
+#### Utility Modules
+
+##### Interning (`intern` module)
+Memory-efficient string and data management through deduplication:
+
+**Code Reference**: [`one_collect/src/intern.rs`](one_collect/src/intern.rs)
+
+###### String Interning
+```rust
+use one_collect::intern::InternedStrings;
+
+// Create an interned strings container
+let mut strings = InternedStrings::new();
+
+// Store strings with deduplication
+let id1 = strings.intern("kernel32.dll");
```

Comment:
Same here - this example is wrong - the function names are incorrect.  Please fix them.